ecological/config.py

```diff
@@ -198,7 +83,61 @@ def from_metaclass_kwargs(cls, metaclass_kwargs: Dict) -> "Options":
             return cls(**options_kwargs)
         except TypeError as e:
             raise ValueError(
-                f"Invalid options for Config class: {metaclass_kwargs}."
+                f"Invalid options for Config class: {options_dict}."
+            ) from e
+
+
+@dataclasses.dataclass
+class Variable:
+    """
+    Represents a single variable from the configuration source
+    and user preferences how to process it.
+    """
+
+    variable_name: Optional[VariableName] = None
+    default: Any = _NO_DEFAULT
+    transform: Optional[TransformCallable] = None
+    source: Optional[Source] = None
+    wanted_type: Type = dataclasses.field(init=False)
+
+    def set_defaults(
+        self,
+        *,
+        variable_name: VariableName,
+        transform: TransformCallable,
+        source: Source,
+        wanted_type: Type,
+    ):
+        """
+        Sets missing properties of the instance of `Variable`` in order to
+        be able to fetch its value with the ``Variable.get`` method.
+        """
+        self.variable_name = self.variable_name or variable_name
+        self.transform = self.transform or transform
+        self.source = self.source or source
+        self.wanted_type = wanted_type
+
+    def get(self) -> VariableValue:
+        """
+        Fetches a value of variable from the ``self.source`` and invoke the
+        ``self.transform`` operation on it. Falls back to ``self.default``
+        if the value is not found.
+        """
+        try:
+            raw_value = self.source[self.variable_name]
+        except KeyError:
+            if self.default is _NO_DEFAULT:
+                raise AttributeError(
+                    f"Configuration error: '{self.variable_name}' is not set."
```

Comment:
```suggestion
                    f"Configuration error: {self.variable_name!r} is not set."
```
I recommend using Python's quoting facilities directly.